api/adapter/s3_adapter.py

```diff
@@ -159,9 +179,18 @@ def delete_dataset_files(
             if self._clean_filename(data_file["Key"]).startswith(raw_file_identifier)
         ]
 
-        files_to_delete.append(
-            {"Key": dataset_metadata.raw_data_path(raw_data_filename)}
-        )
+        self._delete_objects(files_to_delete, raw_data_filename)
+
+    def delete_dataset_files_using_key(self, keys: List[str]):
+        files_to_delete = [{"Key": key} for key in keys]
```

Comment:
You remove this line and pass the objects instead of the keys, you can remove the `_map_object_list_to_key` function